test/lib/processors/nonAsciiEscaper.js

```diff
@@ -0,0 +1,104 @@
+const test = require("ava");
+
+const nonAsciiEscaper = require("../../../lib/processors/nonAsciiEscaper");
+const Resource = require("@ui5/fs").Resource;
+
+/**
+ * Executes string escaping. Returns <code>undefined</code> if nothing was escaped.
+ *
+ * @param {string} input string
+ * @param {Object} [options]
+ * @param {string} encoding character encoding used, e.g. utf8, latin1, ..
+ * @returns {Promise<string|undefined>} escaped string if non-ascii characters present, <code>undefined</code> otherwise
+ */
+const escape = async function(input, options={}, encoding="utf8") {
+	const resource = new Resource({
+		path: "my.properties",
+		buffer: Buffer.from(input, encoding)
+	});
+	return nonAsciiEscaper({resources: [resource], options});
+};
+
+test("Replace ISO-8859-1 characters", async (t) => {
+	t.plan(1);
+
+	const input = `Viele Grüße`;
+	const expected = `Viele Gr\\ufffd\\ufffde`;
+	const [resource] = await escape(input, {}, "latin1");
```

Comment:
The default encoding of the nonAsciiEscaper is utf8, right?
The expected string is not correct.

Reply:
done.

Reply:
add additional tests too